api/repo/update.go

```diff
@@ -238,6 +239,47 @@ func UpdateRepo(c *gin.Context) {
 		}
 	}
 
+	// grab last hook from repo to fetch the webhook ID
+	lastHook, err := database.FromContext(c).LastHookForRepo(r)
+	if err != nil {
+		retErr := fmt.Errorf("unable to retrieve last hook for repo %s: %w", r.GetFullName(), err)
+
+		util.HandleError(c, http.StatusInternalServerError, retErr)
+
+		return
+	}
+
+	// if repo has no hook deliveries, skip webhook update
+	if lastHook.GetWebhookID() != 0 {
+		// if user is platform admin, fetch the repo owner token to make changes to webhook
+		if u.GetAdmin() {
+			// log admin override update repo hook
+			logrus.WithFields(logrus.Fields{
+				"org":  o,
+				"repo": r.GetName(),
+				"user": u.GetName(),
+			}).Infof("platform admin updating repo webhook events for repo %s", r.GetFullName())
```

Comment:
can we move it to _after_ we get the user, because it's not until then, that user is actually getting used?